enzy_htp/structure/structure_region/structure_region.py

```diff
@@ -348,11 +348,16 @@ def convert_to_structure(self, cap_as_residue = True) -> Structure:
         residues: List[Residue] = []
 
         for res in residue_mapper:
+            
+            if isinstance(res, ResidueCap):
+                link_residue = res.link_residue
+
             new_res = deepcopy(res)
             new_res.atoms = residue_mapper[res]
 
             if isinstance(res, ResidueCap):
                 new_res.parent = res.link_residue.parent
+                new_res.link_residue = link_residue
```

Comment:
The new link_residue should be the "new_res" version of the link_residue. That is, the deepcopied one

Reply:
You can create a map that map the old link_residue to the new/copied one. This will be
1. iterate all caps to create the keys
2. during the main loop (this one), after the new_res generation process, if the old one is in the map, put the new one as the value